pkg/promql/engine.go

```diff
@@ -42,19 +42,27 @@ import (
 	"github.com/prometheus/prometheus/storage"
 	"github.com/prometheus/prometheus/util/stats"
 
+	pgmodel "github.com/timescale/promscale/pkg/pgmodel/model"
 	mq "github.com/timescale/promscale/pkg/pgmodel/querier"
 	"github.com/timescale/promscale/pkg/util"
 )
 
 // A Queryable handles queries against a storage.
 // Use it when you need to have access to all samples without chunk encoding abstraction e.g promQL.
 type Queryable interface {
-	// Querier returns a new Querier on the storage.
-	Querier(ctx context.Context, mint, maxt int64) (Querier, error)
+	// SamplesQuerier returns a new promql.Querier on the storage. It helps querying over samples
+	// in the database.
+	SamplesQuerier(ctx context.Context, mint, maxt int64) (SamplesQuerier, error)
+	// ExemplarQuerier returns a new Querier that helps querying exemplars in the database.
+	ExemplarsQuerier(ctx context.Context) ExemplarQuerier
```

Comment:
can we return the querier.ExemplarQuerier here instead of defining a type here

Reply:
Yep, that is done.